### PR TITLE
feat(trends): Add improved/regressed projects to trends

### PR DESCRIPTION
**Pull request by k-fish**
### Summary
This will show widgets to link to the most improved and worst regressed project under the selected filters.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/performance/trends/changedProjects.tsx [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+  justify-content: center;
+  height: 100%;
+`;
+const ProjectTrendContainer = styled('div')`
+  padding: ${space(2)};
+
+  margin-top: ${space(1)};
+  margin-left: ${space(1)};
+
+  font-size: 14px;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/trends/changedProjects.tsx:243]:
```suggestion
  font-size: ${p => p.theme.fontSizeMedium};
```

File: src/sentry/static/sentry/app/views/performance/trends/changedProjects.tsx [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+            {trendPercent}
+          </TrendCirclePrimary>
+          <TrendCircleSecondary>{projectTrend.project}</TrendCircleSecondary>
+        </TrendCircleContent>
+      </TrendCircle>
+    </div>
+  );
+}
+
+const TrendCircle = styled('div')<{color: string}>`
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/trends/changedProjects.tsx:278]: Should we reuse the circleIndicator component here?

File: src/sentry/static/sentry/app/views/performance/trends/utils.tsx [hunk 3]
```diff
@@ -153,15 +155,38 @@ export function transformDeltaSpread(
...
 
+export function getTrendProjectId(
+  trend: NormalizedTrendsTransaction,
+  projects?: Project[]
+): string | undefined;
+
+export function getTrendProjectId(
+  trend: NormalizedProjectTrend,
+  projects?: Project[]
+): string | undefined;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/trends/utils.tsx:166]: This function is defined 3 times.

File: src/sentry/static/sentry/app/views/performance/trends/changedProjects.tsx (hunk 2 above)
Comment (k-fish) [reply src/sentry/static/sentry/app/views/performance/trends/changedProjects.tsx:278]: Not sure about that, I need a circular border with text inside, circle indicator seems to just currently be a circle with a background colour, I don't feel the intent is similar enough to merge them.

File: src/sentry/static/sentry/app/views/performance/trends/utils.tsx (hunk 3 above)
Comment (k-fish) [reply src/sentry/static/sentry/app/views/performance/trends/utils.tsx:166]: I was playing with overloading, I originally overloaded `normalizeTrends` which was helpful but got a little carried away and overloaded this as well, I'll peel this one off but am going to leave it on `normalizeTrends`

**Review by markstory — APPROVED** [review]

**Pull request merged**